.github/workflows/ci-typescript.yml

```diff
@@ -30,7 +30,7 @@ jobs:
       - uses: pnpm/action-setup@v4
         name: Install pnpm
         with:
-          run_install: true
+          run_install: "args: [--frozen-lockfile]"
```

Comment:
This avoids unexpected behaviours.

Reply:
:thinking: this shouldn't be necessary since [it's on by default in CI environments](https://pnpm.io/cli/install#--frozen-lockfile).

If we find it actually is necessary then we should add it in all the github actions where we use `pnpm/action-setup`

Reply:
idk why but enabling this actually gave me the error to understand what's wrong. apparently before that it was installing without any check.

Reply:
hmm could be because of being on pnpm 8.x and not 9.x, I'll see if upgrading to 9.x resolves this...